src/System.Private.CoreLib/src/System/GC.cs

```diff
@@ -635,6 +635,16 @@ public static long GetTotalMemory(bool forceFullCollection)
             return size;
         }
 
+        private static IntPtr _RegisterFrozenSegment(IntPtr sectionAddress, IntPtr sectionSize)
```

Comment:
Why do you need these methods? They are not called by anything...

Reply:
This is the same surface area on CoreCLR, so the generated frozen object assembly is the same across CoreRT and CoreCLR.

Reply:
Does CoreRT let you call these via private reflection like CoreCLR? I would expect it to be disallowed.

Reply:
I think `IgnoresAccessChecksToAttribute` should allow it, no? I haven't tried but let me try it.

  `.custom instance void System.Runtime.CompilerServices.IgnoresAccessChecksToAttribute::.ctor(string) = ( 01 00 16 53 79 73 74 65 6D 2E 50 72 69 76 61 74   // ...System.Privat
                                                                                                          65 2E 43 6F 72 65 4C 69 62 00 00 )                // e.CoreLib..`

Reply:
I'm trying to consume a CoreRT build locally to test this, but I did a code search and I did not find the IgnoreAccessChecksTo being used by CoreRT so it looks like you're right. Then the only option is to make it public in CoreRT, correct?

Reply:
CoreRT compiler does not do build-time access checks.

I thought that you are using private reflection to access these. It sounds like that you emit IL to access these. Is that correct? If you are emitting IL, then it should work ok.

Reply:
```
Microsoft_Experimental_FrozenObjects_InternalCalls_Microsoft_Experimental_FrozenObjects_GC__RegisterFrozenSegment:
  000000014015A850: 55                 push        rbp
  000000014015A851: 48 83 EC 30        sub         rsp,30h
  000000014015A855: 48 8D 6C 24 30     lea         rbp,[rsp+30h]
  000000014015A85A: 48 89 4D 10        mov         qword ptr [rbp+10h],rcx
  000000014015A85E: 48 89 55 18        mov         qword ptr [rbp+18h],rdx
  000000014015A862: 48 8B 4D 10        mov         rcx,qword ptr [rbp+10h]
  000000014015A866: 48 8B 55 18        mov         rdx,qword ptr [rbp+18h]
  000000014015A86A: E8 A1 E1 04 00     call        S_P_CoreLib_System_GC___RegisterFrozenSegment
  000000014015A86F: 48 89 45 F8        mov         qword ptr [rbp-8],rax
  000000014015A873: 48 8B 45 F8        mov         rax,qword ptr [rbp-8]
  000000014015A877: 48 8D 65 00        lea         rsp,[rbp]
  000000014015A87B: 5D                 pop         rbp
  000000014015A87C: C3                 ret
  000000014015A87D: 90                 nop
  000000014015A87E: 90                 nop
  000000014015A87F: 90                 nop
```

```
S_P_CoreLib_System_GC___RegisterFrozenSegment:
  00000001401A8A10: 55                 push        rbp
  00000001401A8A11: 48 83 EC 30        sub         rsp,30h
  00000001401A8A15: 48 8D 6C 24 30     lea         rbp,[rsp+30h]
  00000001401A8A1A: 33 C0              xor         eax,eax
  00000001401A8A1C: 48 89 45 F8        mov         qword ptr [rbp-8],rax
  00000001401A8A20: 48 89 4D 10        mov         qword ptr [rbp+10h],rcx
  00000001401A8A24: 48 89 55 18        mov         qword ptr [rbp+18h],rdx
  00000001401A8A28: 90                 nop
  00000001401A8A29: 48 8B 4D 10        mov         rcx,qword ptr [rbp+10h]
  00000001401A8A2D: 48 8B 55 18        mov         rdx,qword ptr [rbp+18h]
  00000001401A8A31: E8 3A 85 E6 FF     call        RhpRegisterFrozenSegment
  00000001401A8A36: 48 89 45 F0        mov         qword ptr [rbp-10h],rax
  00000001401A8A3A: 48 8B 45 F0        mov         rax,qword ptr [rbp-10h]
  00000001401A8A3E: 48 89 45 F8        mov         qword ptr [rbp-8],rax
  00000001401A8A42: 90                 nop
  00000001401A8A43: EB 00              jmp         00000001401A8A45
  00000001401A8A45: 48 8B 45 F8        mov         rax,qword ptr [rbp-8]
  00000001401A8A49: 48 8D 65 00        lea         rsp,[rbp]
  00000001401A8A4D: 5D                 pop         rbp
  00000001401A8A4E: C3                 ret
  00000001401A8A4F: 90                 nop
```

Reply:
Yup emitting IL. I 💖 No Reflection CoreRT.